pkg/controller/mutate_app_gateway.go

```diff
@@ -59,7 +59,16 @@ func (c AppGwIngressController) GetAppGw() (*n.ApplicationGateway, *appgw.Config
 }
 
 // MutateAppGateway applies App Gateway config.
-func (c AppGwIngressController) MutateAppGateway(appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext) error {
+func (c AppGwIngressController) MutateAppGateway(appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext, invokedForReconciliation bool) error {
+	// if this is a reconciliation task, then before doing anything, compare the current expected state vs cached state of the gateway
+	if invokedForReconciliation {
+		glog.V(5).Info("[reconcile] triggered: ", invokedForReconciliation)
+		if c.configIsSame(appGw) {
```

Comment:
what if GetAppGw is different from cached config?

Reply:
If GetAppGw is different from cached config, then we proceed forward. We use the current gateway state and add back rules that AGIC expects. Since we start by using the current state of the app gateway, our new state has the properties that are part of the current state. Only rules part is overriden.